rollup/internal/controller/sender/sender.go

```diff
@@ -373,13 +396,63 @@ func (s *Sender) resubmitTransaction(tx *gethTypes.Transaction, baseFee uint64)
 		txInfo["adjusted_gas_tip_cap"] = gasTipCap.Uint64()
 		txInfo["original_gas_fee_cap"] = originalGasFeeCap.Uint64()
 		txInfo["adjusted_gas_fee_cap"] = gasFeeCap.Uint64()
+
+	case BlobTxType:
+		originalGasTipCap := tx.GasTipCap()
+		originalGasFeeCap := tx.GasFeeCap()
+		originalBlobGasFeeCap := tx.BlobGasFeeCap()
+
+		// bumping at least 100%
+		gasTipCap := new(big.Int).Mul(originalGasTipCap, big.NewInt(2))
+		gasFeeCap := new(big.Int).Mul(originalGasFeeCap, big.NewInt(2))
+		blobGasFeeCap := new(big.Int).Mul(originalBlobGasFeeCap, big.NewInt(2))
+
+		// adjust for rising basefee
+		currentGasFeeCap := getGasFeeCap(new(big.Int).SetUint64(baseFee), gasTipCap)
+		if gasFeeCap.Cmp(currentGasFeeCap) < 0 {
+			gasFeeCap = currentGasFeeCap
+		}
+
+		// but don't exceed maxGasPrice
+		if gasFeeCap.Cmp(maxGasPrice) > 0 {
+			gasFeeCap = maxGasPrice
+		}
+
+		// gasTipCap <= gasFeeCap
+		if gasTipCap.Cmp(gasFeeCap) > 0 {
+			gasTipCap = gasFeeCap
+		}
+
+		// adjust for rising blobbasefee
+		currentBlobGasFeeCap := getBlobGasFeeCap(new(big.Int).SetUint64(blobBaseFee))
+		if blobGasFeeCap.Cmp(currentBlobGasFeeCap) < 0 {
+			blobGasFeeCap = currentBlobGasFeeCap
+		}
+
+		// but don't exceed maxGasPrice
+		if blobGasFeeCap.Cmp(maxGasPrice) > 0 {
+			blobGasFeeCap = maxGasPrice
```

Comment:
Use the same `maxGasPrice` for execution and blob gas?

Reply:
I reuse the config. wdyt? separating them or use the same one?

Reply:
Let's split this into 2 configs

Reply:
added in [4be6a10](https://github.com/scroll-tech/scroll/pull/1183/commits/4be6a105176a7728e7ab4a10682430087974af19)